api/ingests_api/src/main/scala/uk/ac/wellcome/platform/storage/ingests/api/responses/CreateIngest.scala

```diff
@@ -50,44 +54,73 @@ trait CreateIngest extends ResponseBase with Logging {
       )
     } else {
       triggerIngestStarter(requestDisplayIngest)
+
     }
   }
 
-  private def createBadRequestResponse(description: String): Route =
-    complete(
-      StatusCodes.BadRequest -> UserErrorResponse(
-        contextURL,
-        statusCode = StatusCodes.BadRequest,
-        description = description
+  private def createBadRequestResponse(description: String): Future[Route] =
+    Future {
+      complete(
+        StatusCodes.BadRequest -> UserErrorResponse(
+          contextURL,
+          statusCode = StatusCodes.BadRequest,
+          description = description
+        )
       )
-    )
+    }
 
   private def triggerIngestStarter(
     requestDisplayIngest: RequestDisplayIngest
-  ): Route =
-    ingestStarter.initialise(requestDisplayIngest.toIngest) match {
-      case Success(ingest) =>
-        respondWithHeaders(List(createLocationHeader(ingest))) {
+  ): Future[Route] = {
+    val ingest = requestDisplayIngest.toIngest
+
+    val creationResult = for {
+      trackerResult <- ingestTrackerClient.createIngest(ingest)
+      _ <- trackerResult match {
+        case Right(_) =>
+          Future.fromTry {
+            unpackerMessageSender.sendT(SourceLocationPayload(ingest))
+          }
+        case Left(_) => Future.successful(())
+      }
+    } yield trackerResult
+
+    creationResult
+      .map {
+        case Right(()) =>
+          val headers = List(
+            Location(s"${httpServerConfig.externalBaseURL}/${ingest.id}")
+          )
+
+          respondWithHeaders(headers) {
+            complete(
+              StatusCodes.Created -> ResponseDisplayIngest(
+                ingest,
+                contextURL
+              )
+            )
+          }
+
+        // This will capture both a Conflict and an Internal error from the ingest
+        // tracker.  There's nothing a user can do about either of them, so return
+        // them both as 500 errors.
+        case Left(_) =>
           complete(
-            StatusCodes.Created -> ResponseDisplayIngest(
```

Comment:
Should we pass along the conflict?

Reply:
I did consider it, but I think not.

You get a conflict if you try to create an Ingest and there's already an Ingest with that ID.

The IDs are assigned randomly when we convert a request ingest (from a user) to an internal ingest. Nothing a user does should affect whether you get a conflict here, so I figured better to map them both to 500s and have it show up on our dashboards.